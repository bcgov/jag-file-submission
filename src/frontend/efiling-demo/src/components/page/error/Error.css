.error-info {
  display: flex;
  flex-direction: column;
}
