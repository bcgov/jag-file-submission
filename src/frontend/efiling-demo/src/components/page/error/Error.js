import React from "react";
import PropTypes from "prop-types";
import { Header, Footer, Button } from "shared-components";
import { propTypes } from "../../../types/propTypes";
import "./Error.css";

export default function Error({ page: { header, status, message } }) {
  return (
    <main>
      <Header header={header} />
      <div className="page">
        <div className="content col-md-12">
          <p>
            The following error occurred with your submission. Please click the
            button below to go back home.
          </p>
          <p className="error-info">
            <b>
              Status:&nbsp;
              {status}
            </b>
            <b>
              Message:&nbsp;
              {message}
            </b>
          </p>
          <br />
          <Button
            onClick={() => header.navigate("/")}
            label="Return home"
            styling="bcgov-normal-white btn"
          />
        </div>
      </div>
      <Footer />
    </main>
  );
}

Error.propTypes = {
  page: PropTypes.shape({
    header: propTypes.header,
    status: PropTypes.string.isRequired,
    message: PropTypes.string.isRequired,
  }).isRequired,
};
