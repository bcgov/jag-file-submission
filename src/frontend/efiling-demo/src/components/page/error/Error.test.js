import React from "react";
import { render, getByText, fireEvent } from "@testing-library/react";
import { createMemoryHistory } from "history";
import Error from "./Error";

const header = {
  name: "eFiling Demo Client",
  navigate: createMemoryHistory,
};

const status = "400";
const message = "Could not communicate with CSO.";

const page = { header, status, message };

describe("Error", () => {
  test("Component matches the snapshot", () => {
    const { asFragment } = render(<Error page={page} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("Clicking button on error page redirects to home", () => {
    const { container } = render(<Error page={page} />);

    fireEvent.click(getByText(container, "Return home"));

    expect(window.location.pathname).toEqual("/");
  });
});
