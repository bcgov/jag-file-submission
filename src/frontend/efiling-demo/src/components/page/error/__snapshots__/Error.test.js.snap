// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Component matches the snapshot 1`] = `
<DocumentFragment>
  <main>
    <header>
      <nav
        class="container-fluid navbar navbar-expand-lg navbar-dark"
      >
        <div
          aria-labelledby="title"
          class="navbar-brand pointer"
        />
        <img
          alt="B.C. Government Logo"
          class="img-fluid d-none d-md-block pointer"
          height="44"
          role="presentation"
          src="undefined/images/bcid-logo-rev-en.svg"
          width="181"
        />
        <img
          alt="B.C. Government Logo"
          class="img-fluid d-md-none pointer"
          height="44"
          role="presentation"
          src="undefined/images/bcid-symbol-rev.svg"
          width="64"
        />
        <div
          aria-labelledby="title"
          class="pointer navbar-brand nav-item nav-link"
        />
        <div
          class="navbar-brand"
          id="title"
        >
          eFiling Demo Client
        </div>
      </nav>
    </header>
    <div
      class="page"
    >
      <div
        class="content col-md-12"
      >
        <p>
          The following error occurred with your submission. Please click the button below to go back home.
        </p>
        <p>
          <b>
            Could not communicate with CSO.
          </b>
        </p>
        <br />
        <div>
          <button
            class="normal-white btn"
            data-test-id=""
            type="button"
          >
            Return home
          </button>
        </div>
      </div>
    </div>
    <footer
      class="footer"
    >
      <nav
        class="navbar navbar-expand-sm navbar-dark justify-content-end"
      >
        <button
          class="navbar-toggler"
          data-target="#footerBar"
          data-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse flex-grow-0"
          id="footerBar"
        >
          <ul
            class="navbar-nav text-right"
          >
            <li
              class="nav-item m-1"
            >
              <a
                class="nav-link"
                href="https://www2.gov.bc.ca"
                rel="noopener noreferrer"
                target="_blank"
              >
                BC Government
              </a>
            </li>
            <li
              class="nav-item m-1"
            >
              <a
                class="nav-link"
                href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
                rel="noopener noreferrer"
                target="_blank"
              >
                Disclaimer
              </a>
            </li>
            <li
              class="nav-item m-1"
            >
              <a
                class="nav-link"
                href="https://www2.gov.bc.ca/gov/content/home/privacy"
                rel="noopener noreferrer"
                target="_blank"
              >
                Privacy
              </a>
            </li>
            <li
              class="nav-item m-1"
            >
              <a
                class="nav-link"
                href="https://www2.gov.bc.ca/gov/content/home/accessibility"
                rel="noopener noreferrer"
                target="_blank"
              >
                Accessibility
              </a>
            </li>
            <li
              class="nav-item m-1"
            >
              <a
                class="nav-link"
                href="https://www2.gov.bc.ca/gov/content/home/copyright"
                rel="noopener noreferrer"
                target="_blank"
              >
                Copyright
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </footer>
  </main>
</DocumentFragment>
`;
